paasta_tools/paasta_service_config.py

```diff
@@ -12,29 +12,32 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import logging
-from typing import Any
 from typing import Dict
-from typing import no_type_check
+from typing import Iterable
+from typing import List
+from typing import Tuple
+from typing import Type
+from typing import TypeVar
 
 from service_configuration_lib import read_extra_service_information
 from service_configuration_lib import read_service_configuration
 
-from paasta_tools.adhoc_tools import AdhocJobConfig
-from paasta_tools.chronos_tools import ChronosJobConfig
-from paasta_tools.marathon_tools import MarathonServiceConfig
+from paasta_tools import utils
 from paasta_tools.utils import deep_merge_dictionaries
 from paasta_tools.utils import DEFAULT_SOA_DIR
 from paasta_tools.utils import get_paasta_branch
-from paasta_tools.utils import INSTANCE_TYPES
+from paasta_tools.utils import InstanceConfig
 from paasta_tools.utils import list_clusters
-from paasta_tools.utils import load_deployments_json
 from paasta_tools.utils import load_v2_deployments_json
 
 
 log = logging.getLogger(__name__)
 log.addHandler(logging.NullHandler())
 
 
+_InstanceConfig_T = TypeVar('_InstanceConfig_T', bound=InstanceConfig)
```

Comment:
I've never seen this before. Why do we have to do this kind of thing here and nowhere else?